hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -571,15 +571,20 @@ public final void readData(ObjectDataInput in) throws IOException {
 
     protected abstract void readInternal(ObjectDataInput in) throws IOException;
 
+    protected void toString(StringBuilder sb) {
+    }
+
```

Comment:
Making the toString(StringBuffer) abstract is not good. It forces every Operation implementation to come up with an implementation, even if it is empty. In that case it is better to provide a default implementation (empty) which is good enough for most. It is very important that you always take care of the use of use of an API. If something becomes painful, people will come up with a solution that might be a lot more ugly than you expect.